Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/locate.h

```diff
@@ -0,0 +1,1886 @@
+  std::size_t halfedge_local_index = halfedge_index_in_face(he, tm);
+  coords[halfedge_local_index] = FT(1);
+
+  return std::make_pair(fd, CGAL::make_array(coords[0], coords[1], coords[2]));
+}
+
+/// \brief Returns the location of the given vertex as a `Face_location` in `fd`,
+///        that is an ordered pair composed of `fd` and of the barycentric coordinates
+///        of the vertex in `fd`.
+///
+/// \details If `tm` is the input triangulated surface mesh and given the pair (`f`, `bc`)
+///          such that `bc` is `(w0, w1, w2)`, the correspondance with the weights in `bc`
+///          and the vertices of the face `f` is the following:
+///          - `w0` corresponds to `source(halfedge(f, tm), tm)`
+///          - `w1` corresponds to `target(halfedge(f, tm), tm)`
+///          - `w2` corresponds to `target(next(halfedge(f, tm), tm), tm)`
+///
+/// \tparam TriangleMesh A model of `FaceGraph`
+///
+/// \param vd a vertex of `tm` and a vertex of the face `fd`
+/// \param fd a face of `tm`
+/// \param tm a triangulated surface mesh
+///
+template <typename TriangleMesh>
+typename internal::Locate_types<TriangleMesh>::Face_location
+locate_in_face(const typename boost::graph_traits<TriangleMesh>::vertex_descriptor vd,
+               const typename boost::graph_traits<TriangleMesh>::face_descriptor fd,
+               const TriangleMesh& tm)
+{
+  typedef typename internal::Locate_types<TriangleMesh>::FT               FT;
+
+  FT coords[3] = { FT(0), FT(0), FT(0) };
+  std::size_t vertex_local_index = vertex_index_in_face(vd, fd, tm);
+  coords[vertex_local_index] = FT(1);
+
+  return std::make_pair(fd, CGAL::make_array(coords[0], coords[1], coords[2]));
+}
+
+/// \brief Given a point described by a halfedge `he` and a scalar `t`
+///        as `p = (1 - t) * source(he, tm) + t * target(he, tm)`,
+///        returns this location along the given edge as a `Face_location`, that is
+///        an ordered pair specifying a face containing the location and the
+///        barycentric coordinates of that location in that face.
+///
+/// \details If `tm` is the input triangulated surface mesh and given the pair (`f`, `bc`)
+///          such that `bc` is `(w0, w1, w2)`, the correspondance with the weights in `bc`
+///          and the vertices of the face `f` is the following:
+///          - `w0` corresponds to `source(halfedge(f, tm), tm)`
+///          - `w1` corresponds to `target(halfedge(f, tm), tm)`
+///          - `w2` corresponds to `target(next(halfedge(f, tm), tm), tm)`
+///
+/// \tparam TriangleMesh A model of `FaceGraph`
+///
+/// \param he a halfedge of `tm`
+/// \param t  the parametric distance of the desired point along `he`
+/// \param tm a triangulation surface mesh
+///
+/// \pre `tm` is a triangulated surface mesh.
+///
+template <typename TriangleMesh>
+typename internal::Locate_types<TriangleMesh>::Face_location
+locate_in_face(const typename boost::graph_traits<TriangleMesh>::halfedge_descriptor he,
```

Comment:
no vpm?

Reply:
This is an input with a scalar `t` over a halfedge and thus the barycentric coordinates are `0`, `t`, and `1-t`, regardless of the geometric positions of the vertices.